cfgmgr/natmgrd.cpp

```diff
@@ -98,6 +100,10 @@ void sigterm_handler(int signo)
         natmgr->cleanupMangleIpTables();
         natmgr->cleanupPoolIpTable();
     }
+
+    if (old_sigaction.sa_handler != SIG_IGN && old_sigaction.sa_handler != SIG_DFL) {
```

Comment:
>!= SIG_IGN [](http://example.com/codeflow?start=33&length=10)

Why compare a function pointer to integer macro?

Reply:
SIG_IGN and SIG_DFL are valid values of sa_handler:
https://man7.org/linux/man-pages/man2/sigaction.2.html
```
       sa_handler specifies the action to be associated with signum and
       is be one of the following:

       * SIG_DFL for the default action.

       * SIG_IGN to ignore this signal.

       * A pointer to a signal handling function.  This function
         receives the signal number as its only argument.
```
This comparison ensures that sa_handler value is a valid function pointer and it is safe to call it.